Gems/Atom/Feature/Common/Code/Source/PostProcessing/VignettePass.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <PostProcessing/VignettePass.h>
+#include <PostProcess/PostProcessFeatureProcessor.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Scene.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<VignettePass> VignettePass::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<VignettePass> pass = aznew VignettePass(descriptor);
+            return AZStd::move(pass);
+        }
+
+        VignettePass::VignettePass(const RPI::PassDescriptor& descriptor)
+            : RPI::ComputePass(descriptor)
+        {
+        }
+
+        bool VignettePass::IsEnabled() const
+        {
+            if (!ComputePass::IsEnabled())
+            {
+                return false;
+            }
+            const RPI::Scene* scene = GetScene();
+            if (!scene)
+            {
+                return false;
+            }
+            PostProcessFeatureProcessor* fp = scene->GetFeatureProcessor<PostProcessFeatureProcessor>();
+            const RPI::ViewPtr view = GetRenderPipeline()->GetDefaultView();
+            if (!fp)
+            {
+                return false;
+            }
+            PostProcessSettings* postProcessSettings = fp->GetLevelSettingsFromView(view);
+            if (!postProcessSettings)
+            {
+                return false;
+            }
+            const VignetteSettings* VignetteSettings = postProcessSettings->GetVignetteSettings();
+            if (!VignetteSettings)
+            {
+                return false;
+            }
+            return VignetteSettings->GetEnabled();
+        }
+
+        void VignettePass::FrameBeginInternal(FramePrepareParams params)
+        {
+            // Must match the struct in Vignette.azsl
+            struct Constants
+            {
+                AZStd::array<u32, 2> m_outputSize;
+                AZStd::array<float, 2> m_outputCenter;
+                float m_strength = Vignette::DefaultIntensity;
+            } constants{};
```

Comment:
Add a AZStd::array<float, 3>m_pad at the end to ensure that the struct ends at 16 byte boundary